release-team/hack/sync-bug-triage-github-project-beta.sh

```diff
@@ -0,0 +1,228 @@
+                            }
+                        }) {
+                            projectV2Item {
+                                id
+                            }
+                        }
+                    }' -f project="${PROJECT_ID}" -f item="${1}"  -f type_field="${TYPE_FIELD_ID}" -f option_id="${2}" --silent
+}
+
+echo 'Starting sync...'
+echo -e "[INFO] Fetching the list of open issues and PRs from k/k under the current release milestone, ${MILESTONE}"
+
+## Fetch issues
+MILESTONE_ISSUES_JSON="$(gh api graphql --paginate -f query='
+            query($org: String!, $repo: String!, $milestone: String!){
+              repository(owner: $org, name: $repo) {
+                description
+                url
+                milestones(states: [OPEN],first:1, query: $milestone) {
+                  nodes{
+                    issues(states:[OPEN], first: 200){
+                      nodes{
+                        id
+                      }
+                      pageInfo{
+                        hasNextPage
+                        endCursor
+                      }
+                    }
+                  }
+                  pageInfo{
+                    hasNextPage
+                    endCursor
+                  }
+                }
+              }
+            }' -f org=${ORGANIZATION} -f repo=${REPOSITORY} -f milestone="${MILESTONE}")"
+
+no_of_issues=$(jq ".data.repository.milestones.nodes[].issues.nodes | length" <<< "${MILESTONE_ISSUES_JSON}")
+issues_loop_index_range=$(( no_of_issues - 1 ))
+
+for index in $(seq 0 $issues_loop_index_range);
+do
+	issue_id=$(jq ".data.repository.milestones.nodes[].issues.nodes[$index].id" <<< "${MILESTONE_ISSUES_JSON}")
+	milestone_issue_ids+=("${issue_id}")
+done
+
+## Fetch PRs
+MILESTONE_PRS_JSON="$(gh api graphql --paginate -f query='
+            query($org: String!, $repo: String!, $milestone: String!){
+              repository(owner: $org, name: $repo) {
+                description
+                url
+                milestones(states: [OPEN],first:1, query: $milestone) {
+                  nodes{
+                    pullRequests(states:[OPEN], first: 200){
+                      nodes{
+                        id
+                      }
+                      pageInfo{
+                        hasNextPage
+                        endCursor
+                      }
+                    }
+                  }
+                  pageInfo{
+                    hasNextPage
+                    endCursor
+                  }
+                }
+              }
+            }' -f org=${ORGANIZATION} -f repo=${REPOSITORY} -f milestone="${MILESTONE}")"
+
+no_of_prs=$(jq ".data.repository.milestones.nodes[].pullRequests.nodes | length" <<< "${MILESTONE_PRS_JSON}")
+pr_loop_index_range=$(( no_of_prs - 1 ))
+
+for index in $(seq 0 $pr_loop_index_range);
+do
+  pr_id=$(jq ".data.repository.milestones.nodes[].pullRequests.nodes[$index].id" <<< "${MILESTONE_PRS_JSON}")
```

Comment:
I believe this would need to take into account indexing for the root object when receiving paginated responses as well. I'm not quite sure how to do that in jq off the top of my head, though.

Reply:
addressed. Thank you!